test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -480,3 +480,70 @@
         (fn []
           (let [globals (svcs/shared-globals (get-service svc-utils/*server* :PuppetDBServer))]
             (is (deref (:initial-gc-finished? globals) 5000 false))))))))
+
+(deftest partition-gc-clears-queries-blocking-it-from-getting-accessexclusivelocks
+  (with-unconnected-test-db
+    (let [config (-> (create-temp-config)
+                     (assoc :database *db*)
+                     (assoc-in [:database :gc-interval] "0.01"))
+          store-report #(sync-command-post (svc-utils/pdb-cmd-url)
+                                           example-certname
+                                           "store report"
+                                           cmd-consts/latest-report-version
+                                           (change-report-time example-report %))
+          before-gc (CyclicBarrier. 2)
+          after-gc (CyclicBarrier. 2)
```

Comment:
Are both cyclicbarriers necessary? It looks like in the test we could only wait for after-gc and drop before-gc

Reply:
I think you're right we only need one in this case. That was copy pasta :)